pkg/scalers/metrics_api_scaler.go

```diff
@@ -149,3 +270,52 @@ func (s *metricsAPIScaler) GetMetrics(ctx context.Context, metricName string, me
 
 	return append([]external_metrics.ExternalMetricValue{}, metric), nil
 }
+
+func getMetricAPIServerRequest(meta *metricsAPIScalerMetadata) (*http.Request, error) {
+	var req *http.Request
+	var err error
+
+	if meta.enableAPIKeyAuth {
```

Comment:
How about using switch expression here? It may be easier to extend it in future if we decide to add additional auth method :)

Reply:
Yes, I am already seeing one more use case about JWT token. Completely makes sense. Thanks!